WooCommerce/Classes/View Modifiers/View+NoticesModifier.swift

```diff
@@ -36,6 +36,7 @@ struct NoticeModifier: ViewModifier {
         content
             .overlay(
                 buildNoticeStack()
+                    .padding(.bottom)
```

Comment:
Doesn't this affect the notice when displayed on compact (iPhone) layouts?

Reply:
This is how it looks on iPhone. On iPad (sheet) the problem is that without the padding it is basically on the bottom, so some padding is needed.
Maybe we can check `@Environment(\.presentationMode) var presentationMode` and pass it along to the notice itself to check if we should have padding or not (if modal sheet), what do you think?

![Simulator Screenshot - iPhone 15 Pro - 2024-01-25 at 14 29 04](https://github.com/woocommerce/woocommerce-ios/assets/6242034/dec9dab3-4491-41b4-8652-78b1a2bee5a2)

Reply:
if we can, I think we should avoid adding the extra padding on iPhone layouts.

Reply:
I will dig deeper and find a way!

Reply:
I found a bit more issues now with the notice so I will try to handle all the things in this PR.
When changing split view size/mode it gets messed up:
![Simulator Screenshot - iPad Pro (11-inch) (4th generation) - 2024-01-26 at 10 18 01](https://github.com/woocommerce/woocommerce-ios/assets/6242034/15e8a445-4faf-4523-996f-e560c135cb36)

Reply:
cc @joshheald as I believe he is currently working with the iPad & the Split view project. He may be able to guide you better.

Reply:
I went and did a cleanup/rewrite of the notice modifier to use plain SwiftUI content instead of wrapping with UIKit views etc. That also fixed issues when we resized in split view.
![Simulator Screenshot - iPhone 15 Pro - 2024-01-31 at 11 27 37 copy](https://github.com/woocommerce/woocommerce-ios/assets/6242034/adf22c55-52de-44d4-969f-8e7c12b41aa1)